documentation/library_guide/kernel_templates/esimd/radix_sort_by_key.rst

```diff
@@ -39,7 +40,7 @@ A synopsis of the ``radix_sort_by_key`` function is provided below:
              typename ValuesIterator1, typename KeysIterator2,
              typename ValuesIterator2>
    sycl::event
-   radix_sort_by_key (sycl::queue q, KeysIterator1 keys_first,
+   radix_sort_by_key_copy (sycl::queue q, KeysIterator1 keys_first,
```

Comment:
Should it be `radix_sort_copy_by_key`? I feel it reads slightly more natural.
In the standard, a similar composite suffix is `_copy_if`.

Reply:
I prefer `radix_sort_by_key_copy`, but I'm open for discussion on it.
 
With the current formulation, the name describes the semantics as:  
(radix sort by key) copy          or "sort the data by key, then copy the result"

With the proposed formulation, our intention would be for it to mean: 
(radix sort copy) ... by key

However it also could be read by a user as :
(radix sort) (copy by key) or "radix sort the data, then modify the copy of the data with the key somehow"  possibly it could mean that it copies the sorted data to unique spots based on the key, or masked by key somehow. 

The meaning seems more ambiguous to me based on the punctuation the reader inserts in the second formulation.  Of course, you could also say that a user could read for the first one:  (radix sort) and (by key copy), but that seems much more of a stretch to me.

Reply:
Let's try to make some analogy for existing C++ algorithms with the "copy" flavor. I'll start with `remove` (for which I could imagine `remove_by_key`).

There is `remove(first, last, value);` that removes all elements equal to the value, and `remove_if(first, last, predicate)` that removes all elements for which the predicate evaluates to true. And, there is `remove_copy` and `remove_copy_if` which have an extra parameter for the output sequence where the not-to-be-removed elements are copied.

Now let's imagine `remove_by_key(key_first, key_last, val_first, key)` that removes from both sequences all items with the matching key. In principle, `_by_key` here indicates a condition, similar to how `_if` does. So following the same principle of ordering as for `_if`, I would name a hypothetical out-of-place variant of that as `remove_copy_by_key`.

We can also imagine `unique[_copy]_by_key` that would remove all duplicate keys and corresponding values, and even `copy_by_key` that would copy values which keys match to the given one. And I think in all cases the other order - `remove_by_key_copy`, `unique_by_key_copy`, and especially `by_key_copy` :) - would be worse.

Reply:
Yeah its a fair argument.  `remove_copy_if` is the best example here to convince me.  I think its more straightforward with that API though because `remove_copy` is a much simpler single operation than `radix_sort_copy`.  With the wording of `radix_sort_copy`, is is difficult for me to think of it (semantically) as a single operation to be modified then `by_key`, but rather a pair of operations which happen in sequence.  You are correct though that `radix_sort_copy_by_key` is still the most consistent with the current standard.

I'll make the change.

Reply:
Alright, it should be fixed now.